pkg/operator/encryption/controllers/migration_controller.go

```diff
@@ -223,12 +223,16 @@ func (c *migrationController) migrateKeysIfNeededAndRevisionStable() (migratingR
 		}
 
 		// idem-potent migration start
+		c.eventRecorder.Eventf("EncryptionMigrationStarted", fmt.Sprintf("Started migration for %s", gr.String()))
+		migrationStartedTime := time.Now()
 		finished, result, err := c.migrator.EnsureMigration(gr, ks.Key.Name)
 		if err != nil {
+			c.eventRecorder.Eventf("EncryptionMigrationFailed", fmt.Sprintf("Failed migration of %s (took %s): %v", gr.String(), time.Since(migrationStartedTime), err))
 			errs = append(errs, err)
 			continue
 		}
 		if finished && result != nil {
+			c.eventRecorder.Eventf("EncryptionMigrationFailed", fmt.Sprintf("Finished migration of %s (took %s) with error: %v", gr.String(), time.Since(migrationStartedTime), result))
```

Comment:
would move this into the migrator because you are running into this again and again when result is != nil once.